### ISSUE TITLE
Indentation check incorrectly marks constants defined in an annotation as having the wrong indentation level

### ISSUE THREAD

**Issue by kcoppock** [checkstyle/checkstyle#3302]
Using the IndentationCheck defined as:

```xml
<module name="Indentation">
    <property name="lineWrappingIndentation" value="8"/>
</module>
```

The following code will fail the indentation check
```java
@Retention(RetentionPolicy.SOURCE)
@IntDef({ NamedAnnotation.VAL_1, NamedAnnotation.VAL_2 })
public @interface NamedAnnotation {
    int VAL_1 = 1; // 'member def type'  have incorrect indentation level 8, expected level should be 4. 
    int VAL_2 = 2; // 'member def type'  have incorrect indentation level 8, expected level should be 4. 
}
```

This is using 6.8 but I've also tried 6.19 and it's still reproducible.

**Comment by rnveach** [comment]
@romani Is the expected indentation level of 4 wrong?
The config specifies `lineWrapping`, but this scenario relies on `basicOffset` for where `int` should be located.
Most likely the user is using a tab, instead of 4 spaces, in his code which is why the checker is saying his current indentation is 8.
With no tab definition, we default to 8. https://github.com/checkstyle/checkstyle/blob/9a733ed7985274aadb75dad8bce405749d7e61d6/src/main/java/com/puppycrawl/tools/checkstyle/TreeWalker.java#L67
 Github converts tabs to spaces in issues.
With 4 spaces, or setting the tab width to 4, I get no errors. So this doesn't seem like an issue to me.
````
$ cat TestClass.java
public @interface NamedAnnotation {
	int VAL_1 = 1;
}

$ cat TestConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name = "Checker">

    <module name="TreeWalker">
        <module name="Indentation">
<property name="tabWidth" value="4"/>
<property name="lineWrappingIndentation" value="8"/>
        </module>
    </module>

</module>

$ java -jar checkstyle-6.19-all.jar -c TestConfig.xml TestClass.java
Starting audit...
Audit done.
````

**Comment by kcoppock** [comment]
@rnveach This seems to only happen when the annotation is not the top level class, actually. I can reproduce your success, but it fails if it's nested one level deep, for example:

> public class OuterClass {
>     public @interface InnerInterface {
>         int CONSTANT = 1;
>     }
> }

**Comment by rnveach** [comment]
I can confirm the issue now.
It stems from the fact we have no `AnnotationDefHandler` that behaves like `ClassDefHandler`. (similar to issue #3134)

When looking at the indentation parent hierarchy, it only has parents as `OBJBLOCK`.
[ObjectBlockHandler](https://github.com/checkstyle/checkstyle/blob/924a9d48893fd13082b59b24df53c447a9f10397/src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/ObjectBlockHandler.java#L66-L74) basically only returns the current indentation instead of incrementing it by `basicOffset` which is what `ClassDefHandler` does with [this](https://github.com/checkstyle/checkstyle/blob/924a9d48893fd13082b59b24df53c447a9f10397/src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/AbstractExpressionHandler.java#L94). Because we don't increase the indentation by `basicOffset`, CS expects it to be 4 (the parent's expected indentation level) instead of 8.

**Comment by romani** [comment]
all comments are squashed to remove extra noise..

this case just an example ... that I need to demand from all users strict format of issue report.

**Comment by plan3d** [comment]
@rnveach This issue seems fixed since checkstyle version 8.0

**Comment by strkkk** [comment]
@plan3d can you please share output of running current checkstyle version with parameters from first post?

**Comment by rnveach** [comment]
Referenced issue #3134 in my post was closed with 8.0 milestone, so it does look like that issue resolved this one.

````
$ cat TestClass.java
public class OuterClass {
    public @interface InnerInterface {
        int CONSTANT = 1;
    }
}

$ cat TestConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
<module name="Indentation">
    <property name="lineWrappingIndentation" value="8"/>
</module>
    </module>
</module>

$ java -jar checkstyle-8.40-all.jar -c TestConfig.xml TestClass.java
Starting audit...
Audit done.
````


6.19 and 7.8.2 produced:
````
Starting audit...
[ERROR] TestClass.java:3: 'member def type' have incorrect indentation level 8, expected level should be 4. [Indentation]
Audit done.
````
while starting with 8.0 produced no violations.
